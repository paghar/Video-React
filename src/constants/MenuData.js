//@flow

export type MenuDataType = {
  id: number,
  menuIcon: string,
  menuTitle: string,
  menuLink: string,
};

export const MenuData = [
  { id: 1, menuIcon: '61449', menuTitle: 'Workspace', menuLink: '/home' },
  { id: 2, menuIcon: '61474', menuTitle: 'Projects', menuLink: '/home' },
  { id: 3, menuIcon: '61614', menuTitle: 'Board', menuLink: '/home' },
  { id: 4, menuIcon: '61633', menuTitle: 'Links', menuLink: '/home' },
];
