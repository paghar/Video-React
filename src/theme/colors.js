import theme from 'styled-theming';

export const Generals = {
  i: theme('mode', {
    dark: '#05050d',
  }),
  ii: theme('mode', {
    dark: '#0d0d1a',
  }),
  iii: theme('mode', {
    dark: '#151526',
  }),
  iv: theme('mode', {
    dark: '#212133',
  }),
  v: theme('mode', {
    dark: '#262640',
  }),
  vi: theme('mode', {
    dark: '#7b7b9a',
  }),
  vii: theme('mode', {
    dark: '#b7b7cc',
  }),
  viii: theme('mode', {
    dark: '#ffffff',
  }),
  x: theme('mode', {
    dark: '#e6e6f3',
  }),
  ix: theme('mode', {
    dark: '#f1f1f9',
  }),
};

export const Primary = {
  Dim: theme('mode', {
    dark: '#2172ff',
  }),
  Light: theme('mode', {
    dark: '#317BFC',
  }),
  Dark: theme('mode', {
    dark: '#0E5EE8',
  }),
};

export const PromptWarn = {
  Dim: theme('mode', {
    dark: '#F51653',
  }),
  Light: theme('mode', {
    dark: '#FF4074',
  }),
  Dark: theme('mode', {
    dark: '#DE0540',
  }),
};

export const Secondary = {
  i: theme('mode', {
    dark: '#00d3b2',
  }),
  ii: theme('mode', {
    dark: '#ffcc00',
  }),
  iii: theme('mode', {
    dark: '#f51653',
  }),
  iv: theme('mode', {
    dark: '#64a7ff',
  }),
};

export const Avatars = [
  theme('mode', {
    dark: '#5c4dff',
  }),
  theme('mode', {
    dark: '#685ce6',
  }),
  theme('mode', {
    dark: '#7d73e6',
  }),
  theme('mode', {
    dark: '#928ae6',
  }),
];

export const Gradient = theme('mode', {
  dark:
    'linear-gradient(to bottom, rgba(21, 21, 38, 0), rgba(21, 21, 38, 0.69) 53%, #111120)',
});
