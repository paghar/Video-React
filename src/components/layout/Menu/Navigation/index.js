//@flow
import React from 'react';
import { Logo, Text, IconItem, MenuItem, LogoContainer } from './index.style';
import { Display1, Display3 } from 'theme/fonts';
import { Generals } from 'theme/colors';
import type { MenuDataType } from 'constants/MenuData';

type Props = {
  logoBackgroundColor: string,
  logoImgUrl?: string,
  logoText: string,
  menuData: Array<MenuDataType>,
  logoShortName: string,
};

function Navigation({
  logoBackgroundColor,
  logoImgUrl,
  logoText,
  menuData,
  logoShortName,
}: Props) {
  return (
    <div>
      <LogoContainer>
        <Logo
          backgroundColor={logoBackgroundColor}
          font={Display3}
          imgUrl={logoImgUrl}
          color={Generals.viii}
          width="31px"
          height="31px"
          borderRadius="100%"
        >
          {logoShortName}
        </Logo>

        <Text font={Display1} color={Generals.vii} hoverable>
          {logoText}
        </Text>
      </LogoContainer>

      {menuData.map(item => (
        <MenuItem key={item.id}>
          <IconItem
            fontSize="16px"
            color={Generals.vii}
            hoverColor={Generals.viii}
            iconNumber={item.menuIcon}
          ></IconItem>
          <Text font={Display1} hoverable color={Generals.vi}>
            {item.menuTitle}
          </Text>
        </MenuItem>
      ))}
    </div>
  );
}

export default Navigation;
