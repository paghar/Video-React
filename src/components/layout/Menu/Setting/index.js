//@flow
import React from 'react';
import {
  SettingItems,
  AvatarBtn,
  QuestionBtn,
  SettingBtn,
  ToggleBtn,
} from './index.style';
import { Generals } from 'theme/colors';
import { Display3 } from 'theme/fonts';

type Props = {
  imageAvatar: string,
  isShrinked?: boolean,
  toggleMenu: () => void,
  avatarShortName: string,
  avatarColor: string,
};
function Setting({
  isShrinked,
  toggleMenu,
  imageAvatar,
  avatarShortName,
  avatarColor,
}: Props) {
  return (
    <SettingItems isShrinked={isShrinked}>
      <AvatarBtn
        isShrinked={isShrinked}
        backgroundColor={avatarColor}
        imgUrl={imageAvatar}
        width="30px"
        height="30px"
        borderRadius="100%"
        font={Display3}
        iconColor={Generals.viii}
        color={Generals.viii}
      >
        {avatarShortName}
      </AvatarBtn>

      <QuestionBtn
        isShrinked={isShrinked}
        backgroundColor={Generals.v}
        hoverBackgroundColor={Generals.iv}
        radius="100%"
        icon="61736"
        iconSize="13px"
        iconColor={Generals.vii}
        width="30px"
        height="30px"
      ></QuestionBtn>

      <SettingBtn
        isShrinked={isShrinked}
        backgroundColor={Generals.v}
        hoverBackgroundColor={Generals.iv}
        radius="100%"
        icon="61459"
        iconSize="13px"
        iconColor={Generals.vii}
        width="30px"
        height="30px"
      ></SettingBtn>

      <ToggleBtn
        hoverBackgroundColor={Generals.v}
        isShrinked={isShrinked}
        clicked={() => toggleMenu()}
        backgroundColor={Generals.iii}
        radius="100%"
        icon={isShrinked ? '61497' : '61536'}
        iconSize="16px"
        iconColor={Generals.vii}
        width="30px"
        height="30px"
      ></ToggleBtn>
    </SettingItems>
  );
}

export default Setting;
