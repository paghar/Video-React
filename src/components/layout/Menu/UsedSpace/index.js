//@flow
import React from 'react';
import { LinkDiv, UsedSpaceText } from './index.style';
import { Display3 } from 'theme/fonts';
import { Generals } from 'theme/colors';

type Props = {
  totalSpace: string,
  usedSpace: string,
  className?: string,
};

function usedSpace({ usedSpace, totalSpace, className }: Props) {
  return (
    <div className={className}>
      <UsedSpaceText font={Display3} color={Generals.viii} hoverable>
        {usedSpace} GB of {totalSpace} GB used
      </UsedSpaceText>

      <LinkDiv font={Display3} color={Generals.vi} hoverable>
        Upgrade Now
      </LinkDiv>
    </div>
  );
}

export default usedSpace;
