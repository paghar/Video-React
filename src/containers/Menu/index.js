//@flow
import React, { useState } from 'react';
import Menu from 'components/layout/Menu';
import { MenuData } from 'constants/MenuData';
import logoImage from 'assets/image/avasam-1.png';
import { Avatars } from 'theme/colors';
import imageAvatar from 'assets/image/christopher-campbell-rDEOVtE7vOs-unsplash.png';
import { LEFT_MENU_ANIMATION_DURATION } from 'constants/variables';

type Props = {
  className?: string,
};

function MenuContainer({ className }: Props) {
  const [shrinked, setShrinked] = useState(false);
  const toggleMenu = () => {
    setShrinked(!shrinked);
    // dispatch resize event to update shufflejs after animation done
    setTimeout(() => {
      window.dispatchEvent(new Event('resize'));
    }, LEFT_MENU_ANIMATION_DURATION + 50);
  };
  return (
    <Menu
      isShrinked={shrinked}
      toggleMenu={toggleMenu}
      className={className}
      menuData={MenuData}
      logoBackgroundColor={Avatars[0]}
      logoImgUrl={logoImage}
      logoText="Avasam Agency"
      logoShortName="AA"
      usedSpace="80"
      totalSpace="250"
      imageAvatar={imageAvatar}
      avatarShortName="AB"
      avatarColor={Avatars[0]}
    />
  );
}

export default MenuContainer;
