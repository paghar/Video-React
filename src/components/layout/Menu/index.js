//@flow
import React from 'react';
import {
  MenuContainer,
  BottomContiner,
  UsedSpacePart,
  MenuInnerContent,
} from './index.style';
import Navigation from 'components/layout/Menu/Navigation';
import Setting from 'components/layout/Menu/Setting';
import type { MenuDataType } from 'constants/MenuData';

type Props = {
  isShrinked?: boolean,
  toggleMenu: () => void,
  menuData: Array<MenuDataType>,
  logoBackgroundColor: string,
  logoImgUrl?: string,
  logoText: string,
  usedSpace: string,
  totalSpace: string,
  imageAvatar: string,
  logoShortName: string,
  avatarShortName: string,
  avatarColor: string,
  className?: string,
};

function Menu({
  isShrinked,
  toggleMenu,
  menuData,
  logoBackgroundColor,
  logoImgUrl,
  logoText,
  usedSpace,
  totalSpace,
  imageAvatar,
  logoShortName,
  avatarShortName,
  avatarColor,
  className,
}: Props) {
  return (
    <MenuContainer isShrinked={isShrinked} className={className}>
      <MenuInnerContent>
        <Navigation
          logoBackgroundColor={logoBackgroundColor}
          logoImgUrl={logoImgUrl}
          logoText={logoText}
          menuData={menuData}
          logoShortName={logoShortName}
        ></Navigation>
        <BottomContiner isShrinked={isShrinked}>
          <UsedSpacePart
            usedSpace={usedSpace}
            totalSpace={totalSpace}
          ></UsedSpacePart>
          <Setting
            isShrinked={isShrinked}
            toggleMenu={toggleMenu}
            imageAvatar={imageAvatar}
            avatarShortName={avatarShortName}
            avatarColor={avatarColor}
          ></Setting>
        </BottomContiner>
      </MenuInnerContent>
    </MenuContainer>
  );
}

export default Menu;
