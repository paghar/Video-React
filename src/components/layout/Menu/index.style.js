//@flow
import styled from 'styled-components';
import { Generals } from 'theme/colors';
import UsedSpace from 'components/layout/Menu/UsedSpace';
import { Text as MenuItemText } from './Navigation/index.style';
import { LEFT_MENU_ANIMATION_DURATION } from 'constants/variables';

export const UsedSpacePart = styled(UsedSpace)`
  margin: 0px 0px 0px 23px;
`;

export const MenuContainer = styled.div`
  position: relative;
  transition: width ${LEFT_MENU_ANIMATION_DURATION}ms;
  ${({ isShrinked }) => (isShrinked ? `width: 55px; ` : `width: 208px;`)}
  ${MenuItemText} {
    ${({ isShrinked }) => (isShrinked ? `opacity: 0;` : `opacity: 1;`)}
    transition: opacity ${LEFT_MENU_ANIMATION_DURATION}ms;
  }
  overflow: hidden;
  height: 100vh;
  margin: 0px;
  background-color: ${Generals.iii};
  padding: 0px;
`;

export const MenuInnerContent = styled.div`
  width: 208px;
  position: absolute;
  left: 0;
  top: 0;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  height: 100%;
  margin: 0px;
`;

export const BottomContiner = styled.div`
  ${UsedSpacePart} {
    ${({ isShrinked }) => (isShrinked ? `display: none;` : ``)}
  }
`;
