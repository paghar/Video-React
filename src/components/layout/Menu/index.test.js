//@flow
import React from 'react';
import renderer from 'react-test-renderer';
import Menu from '.';
import { MenuData } from 'constants/MenuData';
import logoImage from 'assets/image/avasam-1.png';
import { Avatars } from 'theme/colors';
import imageAvatar from 'assets/image/christopher-campbell-rDEOVtE7vOs-unsplash.png';

describe('<Menu />', () => {
  it('when menu in open mode', () => {
    const togglerClick = jest.fn();
    const Component = renderer
      .create(
        <Menu
          toggleMenu={() => togglerClick()}
          menuData={MenuData}
          logoBackgroundColor="#fff"
          logoShortName="AA"
          avatarShortName="AB"
          logoImgUrl={logoImage}
          logoText="Avasam Agency"
          usedSpace="80"
          totalSpace="250"
          imageAvatar={imageAvatar}
          avatarColor="#fff"
        ></Menu>
      )
      .toJSON();
    expect(Component).toMatchSnapshot();
  });
});

describe('<Menu />', () => {
  it('when menu in close mode', () => {
    const togglerClick = jest.fn();
    const Component = renderer
      .create(
        <Menu
          isShrinked
          toggleMenu={() => togglerClick()}
          menuData={MenuData}
          logoShortName="AA"
          avatarShortName="AB"
          logoBackgroundColor="#fff"
          logoImgUrl={logoImage}
          logoText="Avasam Agency"
          usedSpace="80"
          totalSpace="250"
          imageAvatar={imageAvatar}
          avatarColor="#fff"
        ></Menu>
      )
      .toJSON();
    expect(Component).toMatchSnapshot();
  });
});
