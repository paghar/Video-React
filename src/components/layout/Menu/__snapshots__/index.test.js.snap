// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> when menu in close mode 1`] = `
.c5 {
  pointer-events: none;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c14 {
  padding-bottom: 8px;
  display: block;
}

.c15 {
  -webkit-text-decoration-line: underline;
  text-decoration-line: underline;
  -webkit-text-decoration-style: solid;
  text-decoration-style: solid;
}

.c3 {
  background-color: #fff;
  background-image: url(avasam-1.png);
  background-size: 100% 100%;
  width: 31px;
  height: 31px;
  border-radius: 100%;
  border: 0px solid #0d0d1a;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  outline: none;
}

.c17 {
  background-color: #fff;
  background-image: url(christopher-campbell-rDEOVtE7vOs-unsplash.png);
  background-size: 100% 100%;
  width: 30px;
  height: 30px;
  border-radius: 100%;
  border: 0px solid #0d0d1a;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  outline: none;
}

.c9 {
  font-family: 'FontAwesome' !important;
  font-size: 16px;
  margin: 0px;
  padding: 0px;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c21 {
  font-family: 'FontAwesome' !important;
  font-size: 13px;
  margin: 0px;
  padding: 0px;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  margin: 21px 0px 50px 12px;
}

.c4 {
  margin: 0px 12px 0px 0px;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c7 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c10 {
  margin: 0px 16px 0px 0px;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c8 {
  padding: 10px 0px 10px 0px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  cursor: pointer;
}

.c8::before {
  content: ' ';
  width: 4px;
  height: 30px;
  margin: 0px 14px 0px 0px;
  visibility: hidden;
}

.c8:hover::before {
  visibility: visible;
}

.c13 {
  margin: 0px 0px 0px 23px;
}

.c26 .c6 {
  opacity: 1;
  -webkit-transition: opacity 300ms;
  transition: opacity 300ms;
}

.c0 {
  position: relative;
  -webkit-transition: width 300ms;
  transition: width 300ms;
  width: 55px;
  overflow: hidden;
  height: 100vh;
  margin: 0px;
  padding: 0px;
}

.c0 .c6 {
  opacity: 0;
  -webkit-transition: opacity 300ms;
  transition: opacity 300ms;
}

.c1 {
  width: 208px;
  position: absolute;
  left: 0;
  top: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  height: 100%;
  margin: 0px;
}

.c11 .c12 {
  display: none;
}

.c22 {
  font-size: 13px;
  display: block;
  padding-left: 0px;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c25 {
  font-size: 16px;
  display: block;
  padding-left: 0px;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c19 {
  direction: ltr;
  outline: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: none;
  background: none;
  width: 30px;
  height: 30px;
  border-radius: 100%;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c16 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin: 0px 12px 29px 12px;
}

.c18 {
  -webkit-order: 3;
  -ms-flex-order: 3;
  order: 3;
}

.c20 {
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
  margin: 0px 0px 16px 0px;
}

.c23 {
  -webkit-order: 2;
  -ms-flex-order: 2;
  order: 2;
  margin: 0px 0px 16px 0px;
}

.c24 {
  -webkit-order: 4;
  -ms-flex-order: 4;
  order: 4;
  margin: 63px 0px 0px 0px;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div>
      <div
        className="c2"
      >
        <button
          className="c3 c4"
          height="31px"
          width="31px"
        />
        <span
          className="c5 c6 c7"
        >
          Avasam Agency
        </span>
      </div>
      <div
        className="c8"
      >
        <span
          className="c9 c10"
          fontSize="16px"
        >
          
        </span>
        <span
          className="c5 c6 c7"
        >
          Workspace
        </span>
      </div>
      <div
        className="c8"
      >
        <span
          className="c9 c10"
          fontSize="16px"
        >
          
        </span>
        <span
          className="c5 c6 c7"
        >
          Projects
        </span>
      </div>
      <div
        className="c8"
      >
        <span
          className="c9 c10"
          fontSize="16px"
        >
          
        </span>
        <span
          className="c5 c6 c7"
        >
          Board
        </span>
      </div>
      <div
        className="c8"
      >
        <span
          className="c9 c10"
          fontSize="16px"
        >
          
        </span>
        <span
          className="c5 c6 c7"
        >
          Links
        </span>
      </div>
    </div>
    <div
      className="c11"
    >
      <div
        className="c12 c13"
      >
        <span
          className="c5 c14"
        >
          80
           GB of 
          250
           GB used
        </span>
        <span
          className="c5 c15"
        >
          Upgrade Now
        </span>
      </div>
      <div
        className="c16"
      >
        <button
          className="c17 c18"
          height="30px"
          width="30px"
        />
        <button
          className="c19 c20"
          height="30px"
          radius="100%"
          width="30px"
        >
          <span
            className="c21 c22"
            fontSize="13px"
          >
            
          </span>
        </button>
        <button
          className="c19 c23"
          height="30px"
          radius="100%"
          width="30px"
        >
          <span
            className="c21 c22"
            fontSize="13px"
          >
            
          </span>
        </button>
        <button
          className="c19 c24"
          height="30px"
          onClick={[Function]}
          radius="100%"
          width="30px"
        >
          <span
            className="c9 c25"
            fontSize="16px"
          >
            
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Menu /> when menu in open mode 1`] = `
.c5 {
  pointer-events: none;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c12 {
  padding-bottom: 8px;
  display: block;
}

.c13 {
  -webkit-text-decoration-line: underline;
  text-decoration-line: underline;
  -webkit-text-decoration-style: solid;
  text-decoration-style: solid;
}

.c3 {
  background-color: #fff;
  background-image: url(avasam-1.png);
  background-size: 100% 100%;
  width: 31px;
  height: 31px;
  border-radius: 100%;
  border: 0px solid #0d0d1a;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  outline: none;
}

.c15 {
  background-color: #fff;
  background-image: url(christopher-campbell-rDEOVtE7vOs-unsplash.png);
  background-size: 100% 100%;
  width: 30px;
  height: 30px;
  border-radius: 100%;
  border: 0px solid #0d0d1a;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  outline: none;
}

.c9 {
  font-family: 'FontAwesome' !important;
  font-size: 16px;
  margin: 0px;
  padding: 0px;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c19 {
  font-family: 'FontAwesome' !important;
  font-size: 13px;
  margin: 0px;
  padding: 0px;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c2 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  margin: 21px 0px 50px 12px;
}

.c4 {
  margin: 0px 12px 0px 0px;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c7 {
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c10 {
  margin: 0px 16px 0px 0px;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c8 {
  padding: 10px 0px 10px 0px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  cursor: pointer;
}

.c8::before {
  content: ' ';
  width: 4px;
  height: 30px;
  margin: 0px 14px 0px 0px;
  visibility: hidden;
}

.c8:hover::before {
  visibility: visible;
}

.c11 {
  margin: 0px 0px 0px 23px;
}

.c0 {
  position: relative;
  -webkit-transition: width 300ms;
  transition: width 300ms;
  width: 208px;
  overflow: hidden;
  height: 100vh;
  margin: 0px;
  padding: 0px;
}

.c0 .c6 {
  opacity: 1;
  -webkit-transition: opacity 300ms;
  transition: opacity 300ms;
}

.c1 {
  width: 208px;
  position: absolute;
  left: 0;
  top: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  height: 100%;
  margin: 0px;
}

.c20 {
  font-size: 13px;
  display: block;
  padding-left: 0px;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c23 {
  font-size: 16px;
  display: block;
  padding-left: 0px;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c17 {
  direction: ltr;
  outline: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: none;
  background: none;
  width: 30px;
  height: 30px;
  border-radius: 100%;
  -moz-user-select: none;
  -khtml-user-select: none;
  -webkit-user-select: none;
  -ms-user-select: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 31px 0px 28px 22px;
}

.c16 {
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
  margin-right: 15px;
}

.c18 {
  -webkit-order: 2;
  -ms-flex-order: 2;
  order: 2;
  margin-right: 15px;
}

.c21 {
  -webkit-order: 3;
  -ms-flex-order: 3;
  order: 3;
  margin-right: 15px;
}

.c22 {
  -webkit-order: 4;
  -ms-flex-order: 4;
  order: 4;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div>
      <div
        className="c2"
      >
        <button
          className="c3 c4"
          height="31px"
          width="31px"
        />
        <span
          className="c5 c6 c7"
        >
          Avasam Agency
        </span>
      </div>
      <div
        className="c8"
      >
        <span
          className="c9 c10"
          fontSize="16px"
        >
          
        </span>
        <span
          className="c5 c6 c7"
        >
          Workspace
        </span>
      </div>
      <div
        className="c8"
      >
        <span
          className="c9 c10"
          fontSize="16px"
        >
          
        </span>
        <span
          className="c5 c6 c7"
        >
          Projects
        </span>
      </div>
      <div
        className="c8"
      >
        <span
          className="c9 c10"
          fontSize="16px"
        >
          
        </span>
        <span
          className="c5 c6 c7"
        >
          Board
        </span>
      </div>
      <div
        className="c8"
      >
        <span
          className="c9 c10"
          fontSize="16px"
        >
          
        </span>
        <span
          className="c5 c6 c7"
        >
          Links
        </span>
      </div>
    </div>
    <div
      className=""
    >
      <div
        className="c11"
      >
        <span
          className="c5 c12"
        >
          80
           GB of 
          250
           GB used
        </span>
        <span
          className="c5 c13"
        >
          Upgrade Now
        </span>
      </div>
      <div
        className="c14"
      >
        <button
          className="c15 c16"
          height="30px"
          width="30px"
        />
        <button
          className="c17 c18"
          height="30px"
          radius="100%"
          width="30px"
        >
          <span
            className="c19 c20"
            fontSize="13px"
          >
            
          </span>
        </button>
        <button
          className="c17 c21"
          height="30px"
          radius="100%"
          width="30px"
        >
          <span
            className="c19 c20"
            fontSize="13px"
          >
            
          </span>
        </button>
        <button
          className="c17 c22"
          height="30px"
          onClick={[Function]}
          radius="100%"
          width="30px"
        >
          <span
            className="c9 c23"
            fontSize="16px"
          >
            
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
